cmd/kubeadm/app/cmd/phases/util_test.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"testing"
+
+	kubeadmapiv1beta1 "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta1"
+	"k8s.io/kubernetes/pkg/version"
+)
+
+func TestSetKubernetesVersion(t *testing.T) {
+
+	ver := version.Get().String()
+
+	tests := []struct {
+		name   string
+		input  string
+		output string
+	}{
+		{
+			name:   "empty version is skipped",
+			input:  "",
+			output: ver,
+		},
+		{
+			name:   "default version is skipped",
+			input:  kubeadmapiv1beta1.DefaultKubernetesVersion,
+			output: ver,
+		},
+		{
+			name:   "any other version is processed",
```

Comment:
Seems like the logic of `SetKubernetesVersion` is "empty version and default version is processed and any other version is skipped". @neolit123 WDYT?

Reply:
sorry, was in a rush and got things mixed up. i will update the PR later today.